### ISSUE TITLE
GraphQL

### ISSUE THREAD

**Issue by drujensen** [johnjansen/crystal-libraries-needed#30]
https://github.com/graphql

**Comment by drujensen** [comment]
@emilebosch is interested in building this.

**Comment by jwoertink** [comment]
Is this not the same thing? https://github.com/ziprandom/graphql-crystal

**Comment by drujensen** [comment]
Good catch @jwoertink should have checked.  I will let Emile know about it.

**Comment by RX14** [comment]
From the graphql-crystal readme it looks like it's simply a graphql parser. That parser could be used to implement a library but i'm sure it's not all too useful by itself.

**Comment by emilebosch** [comment]
Its far from anything at the moment! Ill check out the amount of work that needs to be done

**Comment by drujensen** [comment]
@emilebosch do you want to be assigned to this?

if so, @johnjansen can you add @emilebosch to the ticket?

**Comment by emilebosch** [comment]
The thing is, graphql library is no small feat. I want to take the lead on it, but i need other people that use graphql in their day jobs in order to do it properly.

**Comment by emilebosch** [comment]
But assign it to me, and ill see if i can find some peeps to do it

**Comment by johnjansen** [comment]
hey @emilebosch once you accept the collab, i can assign this to you :-)

**Comment by schovi** [comment]
I can try help. Having fun project which I want port to Crystal from Rails, or at least some parts and GraphQL API is one of them

**Comment by ziprandom** [comment]
Sorry for the late answer, I didn't come across this discussion. actually graphql-crystal is already much more than just a parser. I didn't find the time to update the readme but look at the specs. It already supports the whole query system and introspection. The only mayor protocoll feature that's left is directives which I'm currently working on. directives still is a pretty abstract concept. graphql-crystal already supports the `include` and `exclude` directives but I'm trying to create an api to create custom directives. any contributions are welcome!

**Comment by ziprandom** [comment]
https://github.com/ziprandom/kemal-graphql-example
